### PR TITLE
Update monstergallery

### PR DESCRIPTION
**Pull request by ElisabethSeitz**

### CODE REVIEW

File: frontend/src/contexts/MonsterContextProvider.js [hunk 1]
```diff
@@ -25,12 +25,14 @@ export default function MonsterContextProvider({ children }) {
...
-          updatedMonster,
-        ])
-      )
+      .then((updatedMonster) => {
+        const newState = [...monsters];
+        const updatedMonsterIndex = newState.findIndex(
+          (monster) => monster.id === updatedMonster.id
+        );
+        newState.splice(updatedMonsterIndex, 1, updatedMonster);
+        setMonsters(newState);
```
Comment (jamarob) [inline frontend/src/contexts/MonsterContextProvider.js:34]:
```suggestion
        const newState = monsters.map(monster => monster.id === updatedMonster.id ? updatedMonster : monster)
        setMonsters(newState);
```
short and sweet :)

File: frontend/src/forms/MonsterGallery.js [hunk 2]
```diff
@@ -1,55 +1,48 @@
...
+  savedMonsterImage,
+  imageChangedHandler,
+}) {
+  const images = [
+    '/monsterImages/monster0.jpg',
+    '/monsterImages/monster1.jpg',
+    '/monsterImages/monster2.jpg',
+    '/monsterImages/monster3.jpg',
+    '/monsterImages/monster4.jpg',
+  ];
```
Comment (jamarob) [inline frontend/src/forms/MonsterGallery.js:13]: Can be moved outside of `MonsterGallery` so it is not redefined on every update cycle, just a small performance improvement..

File: frontend/src/forms/MonsterGallery.js [hunk 3]
```diff
@@ -1,55 +1,48 @@
...
+
+  const savedMonsterIndex = savedMonsterImage
+    ? images.indexOf(savedMonsterImage)
+    : -1;
```
Comment (jamarob) [inline frontend/src/forms/MonsterGallery.js:17]: `indexOf` returns -1 if the value is not found, no need for the ternary here.

File: frontend/src/forms/MonsterGallery.js [hunk 4]
```diff
@@ -1,55 +1,48 @@
...
+    setCurrentImage(images[imageIndex]);
+    imageChangedHandler(images[imageIndex]);
+  }, [imageIndex]);
 
   function nextImage() {
-    if (currentImage < amount - 1) {
-      currentImage++;
-      setImage();
-    }
+    setImageIndex((imageIndex + 1) % images.length);
```
Comment (jamarob) [inline frontend/src/forms/MonsterGallery.js:28]: nice wrapping back to the start 👍

File: frontend/src/forms/MonsterGallery.js [hunk 5]
```diff
@@ -1,55 +1,48 @@
...
   }
 
   function previousImage() {
-    if (currentImage > 0) {
-      currentImage--;
-      setImage();
-    }
+    setImageIndex((imageIndex > 0 ? imageIndex : images.length) - 1);
```
Comment (jamarob) [inline frontend/src/forms/MonsterGallery.js:32]: 👍

**Review by jamarob — COMMENTED** [review]
Good job! 🚀

**Review by Asamya — APPROVED** [review]
This looks very good. Nice improvement and refactor of code.

**Pull request merged**